p360monitoring.php

```diff
@@ -0,0 +1,245 @@
+        }
+
+        return $returnHtml;
+    } catch (Exception $e) {
+        Logger::err(__FUNCTION__, $params, $e);
+
+        return $e->getMessage();
+    }
+}
+
+function p360monitoring_CreateAccount(array $params)
+{
+    try {
+        $servers = Params::serverAllowance($params);
+        $websites = Params::websiteAllowance($params);
+        $plan = Plans::getPlanById($params[Params::PLAN_ID]);
+        $kaApi = new KaApi($params);
+        $license = $kaApi->createLicense($plan, (int)$servers, (int)$websites);
+
+        p360monitoring_UpdateModel($params, $license);
+
+        return 'success';
+    } catch (Exception $e) {
+        Logger::err(__FUNCTION__, $params, $e);
+
+        return $e->getMessage();
+    }
+}
+
+function p360monitoring_SuspendAccount(array $params)
+{
+    try {
+        $keyId = $params['customfields']['keyId'];
+        $kaApi = new KaApi($params);
+        $license = $kaApi->suspendLicense($keyId);
+
+        p360monitoring_UpdateModel($params, $license);
+
+        return 'success';
+    } catch (Exception $e) {
+        Logger::err(__FUNCTION__, $params, $e);
+
+        return $e->getMessage();
+    }
+}
+
+function p360monitoring_UnsuspendAccount(array $params)
+{
+    try {
+        $keyId = $params['customfields']['keyId'];
+        $kaApi = new KaApi($params);
+        $license = $kaApi->resumeLicense($keyId);
+
+        p360monitoring_UpdateModel($params, $license);
+
+        return 'success';
+    } catch (Exception $e) {
+        Logger::err(__FUNCTION__, $params, $e);
+
+        return $e->getMessage();
+    }
+}
+
+function p360monitoring_TerminateAccount(array $params)
+{
+    try {
+        $keyId = $params['customfields']['keyId'];
+        $kaApi = new KaApi($params);
+        $license = $kaApi->terminateLicense($keyId);
+
+        p360monitoring_UpdateModel($params, $license);
+
+        return 'success';
+    } catch (Exception $e) {
+        Logger::err(__FUNCTION__, $params, $e);
+
+        return $e->getMessage();
+    }
+}
+
+function p360monitoring_ChangePackage(array $params)
+{
+    try {
+        $keyId = $params['customfields']['keyId'];
+        $servers = Params::serverAllowance($params);
+        $websites = Params::websiteAllowance($params);
+        $plan = Plans::getPlanById($params[Params::PLAN_ID]);
+        $kaApi = new KaApi($params);
+        $license = $kaApi->modifyLicense($keyId, $plan, (int)$servers, (int)$websites);
+
+        p360monitoring_UpdateModel($params, $license);
+
+        return 'success';
+    } catch (Exception $e) {
+        Logger::err(__FUNCTION__, $params, $e);
+
+        return $e->getMessage();
+    }
+}
+
+function p360monitoring_UpdateModel($params, $license)
+{
+    $params['model']->serviceProperties->save([
```

Comment:
undefined index?

Reply:
If called from the module hook functions, as it is now, then no